mcnode/node.go

```diff
@@ -323,3 +343,106 @@ func (node *Node) doShutdown() {
 	node.ds.Close()
 	os.Exit(0)
 }
+
+func (auth *PeerAuth) fromJSON(rmap map[string]interface{}) error {
+	auth.mx.Lock()
+	defer auth.mx.Unlock()
```

Comment:
this just executes when it goes out of scope? weird. I guess it's like a `finally` a bit?